packages/hebao_v1/contracts/modules/core/ERC1271Module.sol

```diff
@@ -46,13 +52,27 @@ abstract contract ERC1271Module is ERC1271, SecurityModule
         override
         returns (bytes4 magicValue)
     {
-        address wallet = msg.sender;
-        if (securityStore.isLocked(wallet)) {
-            return 0;
-        }
+        uint8 sigType = _signature.toUint8(0);
+
+        if (sigType == uint8(SignType.DEFAULT)) {
+            address wallet = msg.sender;
+            if (securityStore.isLocked(wallet)) {
+                return 0;
+            }
+
+            address _owner = Wallet(wallet).owner();
+            bytes memory _sig = _signature.slice(8, _signature.length - 8);
+            return _signHash.verifySignature(_owner, _sig) ? ERC1271_MAGICVALUE : bytes4(0);
+        } else if (sigType == uint8(SignType.PERPETUAL)) {
+            uint ownerIdx = _signature.toUint32(8);
```

Comment:
I put ownerIdx here so we don't have to do a binary search in contract for the signing owner address.